usr/lib/ep11_stdll/ep11_specific.c

```diff
@@ -6572,6 +6572,28 @@ static CK_RV dh_generate_keypair(STDLL_TokData_t *tokdata,
         goto dh_generate_keypair_end;
     }
 
+    /* Supply CKA_VALUE_BITS to private key if not present or zero */
+    if (template_attribute_get_ulong(priv_tmpl, CKA_VALUE_BITS,
+                                     &value_bits) != CKR_OK ||
+        value_bits == 0) {
+        value_bits = data_len; /* private key is same length as public key */
```

Comment:
I think data_len is a byte length, but value_bits is a bit length -> value_bits = data_len * 8 ?

Reply:
Ah you are right, fixed.